Fixes unauthorized error in es datasources

cqliu1 opened the pull request:
Closes #24461.

The issue was that our check for if security was enabled evaluated to true even when using a basic license. This provides a better check to determine if security is enabled or not. This is just a temporary fix until security exposes a method on the plugin that we should replace our helper function with (see #24616).

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/6564/)

w33ble commented on x-pack/plugins/canvas/server/lib/feature_check.js:
```diff
@@ -0,0 +1,17 @@
+    securityPlugin &&
+    xpackInfo.isAvailable() &&
+    xpackInfo.feature('security').isEnabled() &&
+    !xpackInfo.license.isOneOf('basic')
```
Curious, is this really required? Won't `xpackInfo.feature('security').isEnabled()` be false if the license is wrong?

cqliu1 replied:
So if you run elasticsearch with a basic license and 'xpack.security.enabled: true`, `server.xpack_main.feature('security').isEnabled()` will be true, but `server.xpack_main.feature('security').isAvailable()` will be false. Instead of checking the license, I can check `isAvailable()`, but I do need both checks in case you have a non-basic license but have security disabled.

cqliu1 commented:
@clintandrewhall @w33ble So I realized I broke our unit tests for `create_handler`. I fixed the unit tests and cleaned up the logic for checking if security is enabled.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/6699/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/6705/)

cqliu1 commented:
> Are there constants for the license types for the unit test?

Not that I'm aware of. I can certainly add some contents. It might actually be better to change the variable to a boolean `isSecurityAvailable` instead of using `license` because what I'm really concerned with if security is available as a plugin or not.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/6868/)

cqliu1 commented:
retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/6903/)
